### PR TITLE
chore(detail): remove classic theme

### PR DESCRIPTION
**Pull request by DipperTheDan**
removes classic theme stories and old documentation. This is replaced with CSF and MDX examples of
the detail component.

fixes FE-3137

### Proposed behaviour
Removal of `classic theme` stories from the Detail component. 
The documentation folder has been deleted.
StoriesOf story has been refactored to be in CSF.
An MDX example for Detail has been created.

### Current behaviour
Detail has a `classic theme` story.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
N/A

### Testing instructions
- `git checkout FE-3137-remove_classic_theme_detail`
- Start storybook by using `npm start`
- Navigate to the Detail story in Storybook
- No `classic` story should be present.
- Detail should have an MDX example that documents the component.

### CODE REVIEW

File: src/components/detail/detail.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    </Detail>
+  </Story>
+</Preview>
+
+### Detail with a Footnote
+Detail can also have a footnote by using the `footnote` prop.
+<Preview>
+  <Story name="with Footnote">
+    <Detail
+    footnote="This is a footnote."
```
Comment (mkrds) [inline src/components/detail/detail.stories.mdx:45]: Same here, wrong indent, just make it 1 line

File: src/components/detail/detail.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+</Preview>
+
+### Inside of Components
+Detail can be rendered inside of components. Inside of this example it is renendered inside of a `Card` component
+
+<Preview>
+  <Story name="inside of card">
+    <Card width='300px'>
+      <div style={{paddingTop: '16px'}}>
+        <Detail
```
Comment (mkrds) [inline src/components/detail/detail.stories.mdx:73]: +1

File: src/components/detail/detail.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        >
+          This example of Detail just has children.
+        </Detail>
+      </div>
+      <Hr />
+      <div style={{paddingTop: '8px'}}>
+        <Detail
```
Comment (mkrds) [inline src/components/detail/detail.stories.mdx:80]: +1

File: src/components/detail/detail.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    </Card>
+  </Story>
+</Preview>
+
+In this example Detail is inside of the `Tile` component.
+<Preview>
+  <Story name="inside of tile">
+    <Tile width={60}>
+      <div style={{paddingTop: '16px'}}>
+        <Detail
```
Comment (mkrds) [inline src/components/detail/detail.stories.mdx:104]: +1

File: src/components/detail/detail.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        >
+          This example of Detail just has children.
+        </Detail>
+      </div>
+      <div style={{paddingTop: '8px'}}>
+        <Detail
```
Comment (mkrds) [inline src/components/detail/detail.stories.mdx:110]: +1

**Review by mkrds — DISMISSED** [review]

File: src/components/detail/detail.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+<Preview>
+  <Story name="with icon">
+    <Detail icon='bin' footnote='This is a footnote' >
+      This is where the children will live.
+    </Detail>
+  </Story>
+</Preview>
+
+### Inside of Components
+Detail can be rendered inside of components. Inside of this example it is renendered inside of a `Card` component
```
Comment (sjohnson-sage) [inline src/components/detail/detail.stories.mdx:62]: Typo on `rendered` here

File: src/components/detail/detail.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+<Preview>
+  <Story name="inside of card">
+    <Card width='300px'>
+      <div style={{paddingTop: '16px'}}>
+        <Detail
+        >
```
Comment (sjohnson-sage) [inline src/components/detail/detail.stories.mdx:69]: May as well move this closing bracket (if bracket is the right term) up the the same line as `<Detail>`

File: src/components/detail/detail.stories.mdx (hunk 6 above)
Comment (DipperTheDan) [reply src/components/detail/detail.stories.mdx:62]: Well spotted.

**Review by sjohnson-sage — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by sjohnson-sage — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by mkrds — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]
Good job with `cypress` stuff

**Review by sjohnson-sage — APPROVED** [review]

**Pull request merged**